Framework/Beamline/inc/MantidBeamline/DetectorInfo.h

```diff
@@ -57,13 +57,17 @@ namespace Beamline {
 class MANTID_BEAMLINE_DLL DetectorInfo {
 public:
   DetectorInfo(const size_t numberOfDetectors);
+  DetectorInfo(const size_t numberOfDetectors,
+               const std::vector<size_t> &monitorIndices);
 
   size_t size() const;
 
+  bool isMonitor(const size_t index) const;
   bool isMasked(const size_t index) const;
   void setMasked(const size_t index, bool masked);
```

Comment:
Currently Mantid does not allow marking detectors as monitors in parameterized instrument. That is, once the instrument is in a workspace it is not possible to mark detectors as monitors, in can only be done at creation of the *base* instrument.

We might add `setMonitor` if required at a later time.